### PR TITLE
data.load() must be private

### LINKED ISSUE #3947: should data_dir.load be a public function
https://github.com/scikit-image/scikit-image/blob/822ad3edddaba91e43dfc1adf2fe4339dbefb97b/skimage/data/__init__.py#L23

it seems like it is mostly an internal function. In migrating to pooch, ~~I would like to really add some hacks that make it really only for our internal data~~
https://github.com/scikit-image/scikit-image/pull/3945

### PR DESCRIPTION
**Pull request by sciunto**
## Description

Fixes #3947 

Cc @hmaarrfk 

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/data/__init__.py [hunk 1]
```diff
@@ -52,6 +53,29 @@
...
+    Returns
+    -------
+    img : ndarray
+        Image loaded from ``skimage.data_dir``.
+
+    Note
+    ----
+    This functions is deprecated and will be removed in 0.18.
+    """
+    warn('This function is deprecated and will be removed in 0.18', stacklevel=2)
```
Comment (hmaarrfk) [inline skimage/data/__init__.py]:
```suggestion
    warn('This function is deprecated and will be removed in 0.18. '
         'Use `skimage.io.load` or `imageio.imread` directly.', 
         stacklevel=2)
```

File: skimage/data/__init__.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 import os as _os
 
 import numpy as _np
+from warnings import warn
 
 from .._shared._warnings import expected_warnings, warn
```
Comment (hmaarrfk) [inline skimage/data/__init__.py]: lets remove this line? when did it sneak in....

Comment (sciunto) [reply skimage/data/__init__.py]: ohoh!

**Comment by sciunto** [comment]
Thanks @jni and @hmaarrfk

**Pull request merged**